File: frontend/src/models/levelModel.ts
import type BaseModel from "./baseModel";

export default interface LevelModel extends BaseModel {
    level: number;
    levelType: LevelTypeEnum;

    value: number;
    availableCoin: number;
}
export enum LevelTypeEnum {
    LimitEnergyLevel = 0,
    MultipleClickLevel = 1,
    RechargeEnergyLevel = 2,
    RoBotLevel = 3,
};

File: backend/Common/Domain/Models/LevelModel.cs
using Presentation.Common.Domain.Enums;

namespace Presentation.Common.Domain.Models;
public class LevelModel : BaseModel
{
    public int Level { get; set; }
    public int Value { get; set; }
    public long AvailableCoin { get; set; }

    public LevelTypeEnum LevelType { get; set; }
}
